new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -193,18 +193,20 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
             numPagesRef.current
         );
 
-        if (newPageNum === pageNumRef.current) {
-            // Don't need to switch pages so just update `logEventNum` in the URL.
-            updateLogEventNumInUrl(numEvents, logEventNumRef.current);
-        } else if (STATE_DEFAULT.pageNum !== pageNumRef.current) {
-            // This is not the initial page load, so request a page switch.
-            // NOTE: We don't need to call `updateLogEventNumInUrl()` since it's called when
-            // handling the `WORKER_RESP_CODE.PAGE_DATA` response (the response to
-            // `WORKER_REQ_CODE.LOAD_PAGE` requests) .
-            mainWorkerPostReq(WORKER_REQ_CODE.LOAD_PAGE, {
-                cursor: {code: CURSOR_CODE.PAGE_NUM, args: {pageNum: newPageNum}},
-                decoderOptions: getConfig(CONFIG_KEY.DECODER_OPTIONS),
-            });
+        if (STATE_DEFAULT.pageNum !== pageNumRef.current) {
+            if (newPageNum === pageNumRef.current) {
+                // Don't need to switch pages so just update `logEventNum` in the URL.
+                updateLogEventNumInUrl(numEvents, logEventNumRef.current);
+            } else {
```

Comment:
```suggestion
        if (STATE_DEFAULT.pageNum !== pageNumRef.current) {
            return;
        } else if (newPageNum === pageNumRef.current) {
            // Don't need to switch pages so just update `logEventNum` in the URL.
            updateLogEventNumInUrl(numEvents, logEventNumRef.current);
        } else {
```

Reply:
Hmm... How about
```suggestion
        if (STATE_DEFAULT.pageNum !== pageNumRef.current) {
            pageNumRef.current = newPageNum;

	    return;
	} else if (newPageNum === pageNumRef.current) {
	    // Don't need to switch pages so just update `logEventNum` in the URL.
	    updateLogEventNumInUrl(numEvents, logEventNumRef.current);
	} else {
```
? i.e. update `pageNumRef.current` before returning.

Reply:
I changed the `!==` to `===`, will that save `pageNumRef.current = newPageNum;`?

Reply:
Makes sense. There's no update `pageNumRef.current = newPageNum` if they're the same.

Reply:
Sorry for misreading the code. I think we still need the `pageNumRef.current = newPageNum` update before returning, right? Say what if `newPageNum !== STATE_DEFAULT.pageNum`?